### PR TITLE
Add data table component

### LINKED ISSUE #90: Feat: Create (data) Table component
## Description

as part of the next milestone, we need to add the table component.

[Figma Design](https://www.figma.com/file/msxmKajRPLYiDDmsP60Ndh/Design-System-v1.1?type=design&node-id=597-48013&mode=dev)

## Tasks

should support:

- [ ] multiple selection
- [ ] single selection
- [ ] pagination
- [ ] dense cells
- [ ] sorting
- [ ] light / dark mode

### PR DESCRIPTION
**Pull request by tewshi**
Closes #90

### CODE REVIEW

File: src/components/tables/DataTable.stories.ts [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+export const Dense: Story = {
+  args: {
+    rows: data,
+    dense: true,
+  },
+};
+
+export const Loading: Story = {
+  args: {
+    rows: [],
```
Comment (lukicenturi) [inline src/components/tables/DataTable.stories.ts:225]: rows: data

without this the storybook section become empty

Comment (tewshi) [reply src/components/tables/DataTable.stories.ts:225]: thanks, I missed this...

Comment (tewshi) [reply src/components/tables/DataTable.stories.ts:225]: I see the issue, when columns is not passed, no data to generate the columns as rows are empty...
I intentionally wanted the table to show only loading state and headers, without body, you can check again.

**Review by kelsos — APPROVED** [review]
lgtm

**Pull request merged**